src/MCPClient/lib/clientScripts/sanitize_object_names.py

```diff
@@ -20,140 +20,251 @@
+        for event in Event.objects.filter(
+            file_uuid__in=[event.file_uuid for event in events]
+        ):
+            event.agents.add(*event_agents)
+
+        if len(self.files_index) > 0:
+            logger.info("Sanitized batch of %s files", len(self.files_index))
+
+            self.files_index = {}
         else:
-            logger.info("No sanitization for %s", current_location)
-            job.pyprint("No sanitization found for", current_location)
+            logger.info("No file sanitization required.")
+
+    def apply_dir_updates(self):
+        """
+        Run a single batch of Directory updates.
+        """
+        if self.directory_queryset is None:
+            logger.info("No directory sanitization required.")
+            return
+
+        # We pass through _all_ objects here, as they may not be normalized in
+        # the db :(
+        for dir_obj in self.directory_queryset.iterator():
+            old_location = unicodedata.normalize("NFC", dir_obj.currentlocation)
+            try:
+                sanitized_location = self.dirs_index[old_location]
+            except KeyError:
+                continue
+
+            dir_obj.currentlocation = sanitized_location
+            dir_obj.save()
+
+            # TODO: Dir sanitizations don't generate events?
+            # Is seems like they should.
+
+        if len(self.dirs_index) > 0:
+            logger.info("Sanitized batch of %s directories", len(self.dirs_index))
+            self.dirs_index = {}
+        else:
+            logger.info("No directory sanitization required.")
+
+    def add_file_to_batch(self, old_path, new_path):
+        """
+        Index a path change for later updating in the database.
 
-    return 0
+        If our index is over a certain size, then trigger application of the
+        batched changes.
+        """
+        old_path = self.normalize_path_for_db(old_path)
+        new_path = self.normalize_path_for_db(new_path)
+
+        self.files_index[old_path] = new_path
+
+        if len(self.files_index) >= self.BATCH_SIZE:
+            self.apply_file_updates()
+
+    def add_dir_to_batch(self, old_path, new_path):
+        """
+        Index a path change for later updating in the database.
+
+        If our index is over a certain size, then trigger application of the
+        batched changes.
+        """
+        old_path = self.normalize_path_for_db(old_path)
+        new_path = self.normalize_path_for_db(new_path)
+
+        # Add trailing slashes if necessary
+        old_path = os.path.join(old_path, "")
+        new_path = os.path.join(new_path, "")
+
+        self.dirs_index[old_path] = new_path
+
+        if len(self.dirs_index) >= self.BATCH_SIZE:
+            self.apply_dir_updates()
+
+    def sanitize_objects(self):
+        """
+        Iterate over the filesystem, sanitizing as we go. Updates made on disk
+        are batched and then applied to the datab
```

Comment:
“applied to the datab” – is this sentence incomplete?

Reply:
Whoops! Thanks, I'll finish that thought.